src/components/Breed/BreedList.js

```diff
@@ -0,0 +1,58 @@
+import React, { useState, useEffect } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import BreedButton from "./BreedButton";
+import { getAllBreeds } from "../../store/breed-actions";
+import { getBreedImages } from "../../store/breed-image-actions";
+
+const BreedList = () => {
+  const dispatch = useDispatch();
+
+  const breeds = useSelector((state) => state.breed.breeds);
+  const loading = useSelector((state) => state.breed.loading);
+  const errors = useSelector((state) => state.breed);
+
+  const [activeBtn, setActiveBtn] = useState("");
+  const [breedList, setBreedList] = useState([]);
+
+  useEffect(() => {
+    dispatch(getAllBreeds());
+  }, [dispatch]);
+
+  useEffect(() => {
+    setBreedList(breeds);
+  }, [breeds, dispatch]);
+
+  const breedClickHandler = (event) => {
+    setActiveBtn(event.target.value);
+    dispatch(getBreedImages(event.target.value));
+  };
+
+  let content = <p>No Records found</p>;
+
+  if (!loading && errors.isError) {
+    content = <p>{errors.errors}</p>;
```

Comment:
You can return the content instead of assign it to a variable. Something like this 
```tsx
if(errors.isError){
  return <p>No record found</p>
}
if(loading){
  return <p>Loading</p>
}
if(breedList.length !== 0){
  return (
     <BreadButton 
        key={data} 
        onClick={breedClickHandler}
        value={data}
        primary={data === activeBtn ? true : false} >
          {data}
        </BreedButton>
  )


}
```

Reply:
This is Done